src/manifest/store.rs

```diff
@@ -335,23 +387,21 @@ impl StoredManifest {
         }
 
         let manifest = Manifest::cloned(parent_db, parent_manifest);
-        self.update_manifest(manifest).await
+        let dirty = DirtyManifest::new(self.id, self.next_id(), manifest);
+        self.update_manifest(dirty).await
     }
 
-    pub(crate) async fn update_db_state(&mut self, core: CoreDbState) -> Result<(), SlateDBError> {
-        let manifest = Manifest {
-            parent: self.manifest.parent.clone(),
-            core,
-            writer_epoch: self.manifest.writer_epoch,
-            compactor_epoch: self.manifest.compactor_epoch,
-        };
-        self.update_manifest(manifest).await
-    }
-
-    async fn update_manifest(&mut self, manifest: Manifest) -> Result<(), SlateDBError> {
-        let new_id = self.id + 1;
+    pub(crate) async fn update_manifest(
+        &mut self,
+        manifest: DirtyManifest,
+    ) -> Result<(), SlateDBError> {
+        let new_id = self.next_id();
+        if manifest.next_id() != new_id {
+            return Err(ManifestVersionExists);
+        }
+        let manifest = manifest.into();
         self.manifest_store
-            .write_manifest(new_id, &manifest)
+            .write_manifest(self.next_id(), &manifest)
```

Comment:
🤔 Should this be `new_id`?

Reply:
Yeah might be easier to understand. It's still correct though, since `new_id` is set using `next_id()`.

Reply:
Ah, I thought next_id advanced the ID as well, not simply a reader. My bad.

Reply:
.. but yea.. would be more readable :P